### PR TITLE
Feature/9 initial cicd pipeline

### LINKED ISSUE #9: Setup CI / CD pipelines
Initial setup of pipeline to be refined while the project is developing.

### PR DESCRIPTION
**Pull request by luccalb**
Create an initial pipeline with following structure:

- License Check
  - setup python & install deps
  - export all direct and transient dependencies to txt file
  - check the txt file and fail on copyleft licenses 
- Lint & Test
  - setup python & install deps
  - lint using `flake8`
  - test using `pytest`

Closes #9

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
```
Comment (felix-zailskas) [inline .github/workflows/python-app.yml:26]: Should the python Version be 3.10 everywhere?

**Review by felix-zailskas — CHANGES_REQUESTED** [review]
Looks good. Just a small comment towards the python versions.

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (luccalb) [reply .github/workflows/python-app.yml:26]: I think so too, classic copy paste mistake. Thanks for pointing it out!

Comment (felix-zailskas) [reply .github/workflows/python-app.yml]: Feel free to resolve this comment once it is changed.

**Review by felix-zailskas — APPROVED** [review]
Thanks for the change!

**Pull request merged**